### PR TITLE
add job

### LINKED ISSUE #83: Add background job
Hangfire

### PR DESCRIPTION
**Pull request by SergiLomtadze**
closes #83

### CODE REVIEW

File: src/ExadelMentorship.Jobs/IWeatherJob.cs [hunk 1]
```diff
@@ -2,6 +2,6 @@
 {
     public interface IWeatherJob
     {
-        Task HistorySaving();
+        void HistorySaving();
```
Comment (kdavletov-exadel) [inline src/ExadelMentorship.Jobs/IWeatherJob.cs:5]: Mind the naming convention for methods.

File: src/ExadelMentorship.WebApi/Controllers/WeatherController.cs [hunk 2]
```diff
@@ -45,5 +48,11 @@ public Task<IEnumerable<City>> GetFutureWeather([FromRoute] string cityName, [Fr
                 }
             );
         }
+
+        public string EnableJobs()
+        {
+            _weatherJob.HistorySaving();
```
Comment (kdavletov-exadel) [inline src/ExadelMentorship.WebApi/Controllers/WeatherController.cs:54]: This probably should trigger during app start and most likely be idempotent

**Review by kdavletov-exadel — CHANGES_REQUESTED** [review]

File: src/ExadelMentorship.Jobs/IWeatherJob.cs (hunk 1 above)
Comment (SergiLomtadze) [reply src/ExadelMentorship.Jobs/IWeatherJob.cs:5]: done

File: src/ExadelMentorship.WebApi/Controllers/WeatherController.cs (hunk 2 above)
Comment (SergiLomtadze) [reply src/ExadelMentorship.WebApi/Controllers/WeatherController.cs]: done

File: src/ExadelMentorship.Jobs/WeatherJob.cs [hunk 3]
```diff
@@ -1,10 +1,42 @@
...
+using ExadelMentorship.DataAccess;
+using Hangfire;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Options;
+
+namespace ExadelMentorship.WebApi.Jobs
 {
-    public class WeatherJob : IWeatherJob
+    public class WeatherJob : IHostedService
```
Comment (kdavletov-exadel) [inline src/ExadelMentorship.Jobs/WeatherJob.cs:11]: Use [`BackgroundService`](https://www.google.com/search?client=firefox-b-d&q=backgroundservice+vs+ihostedservice) instead

File: src/ExadelMentorship.Persistence/PersistenceServiceExtentions.cs [hunk 4]
```diff
@@ -13,7 +13,7 @@ public static void AddPersistenceServices(this IServiceCollection services)
             services.AddDbContext<ApplicationDbContext>((s, option) =>
                 option.UseSqlServer(s.GetRequiredService<IConfiguration>().GetConnectionString("DefaultConnection")));
 
-            services.AddScoped<IWeatherHistorySavingRepository, WeatherHistorySavingRepository>();
+            services.AddScoped<IWeatherHistorySavingRepository, WeatherHistorySavingRepository>();
```
[the remark this answers is not in the document]
Comment (SergiLomtadze) [reply src/ExadelMentorship.Persistence/PersistenceServiceExtentions.cs:16]: done

**Comment by SergiLomtadze** [comment]
no, I have not touched CI file

File: src/ExadelMentorship.Jobs/WeatherJob.cs (hunk 3 above)
Comment (SergiLomtadze) [reply src/ExadelMentorship.Jobs/WeatherJob.cs]: done

**Review by kdavletov-exadel — CHANGES_REQUESTED** [review]
Did you remove the actual `webapi.log` file from the PR branch?

**Review by kdavletov-exadel — APPROVED** [review]
Now looks neat!

**Pull request merged**